src/BoundingBoxCameraSensor.cc

```diff
@@ -0,0 +1,525 @@
+
+  /// \brief RGB Image to draw boxes on it
+  public: rendering::Image image;
+
+  /// \brief Buffer contains the BoundingBox map data
+  public: unsigned char *imageBuffer {nullptr};
+
+  /// \brief Connection to the new BoundingBox frames data
+  public: common::ConnectionPtr newBoundingBoxConnection;
+
+  /// \brief Connection to the Manager's scene change event.
+  public: common::ConnectionPtr sceneChangeConnection;
+
+  /// \brief Just a mutex for thread safety
+  public: std::mutex mutex;
+
+  /// \brief BoundingBoxes type
+  public: rendering::BoundingBoxType type
+    {rendering::BoundingBoxType::VisibleBox2D};
+};
+
+//////////////////////////////////////////////////
+BoundingBoxCameraSensor::BoundingBoxCameraSensor()
+  : CameraSensor(), dataPtr(new BoundingBoxCameraSensorPrivate)
+{
+}
+
+/////////////////////////////////////////////////
+BoundingBoxCameraSensor::~BoundingBoxCameraSensor()
+{
+  if (this->dataPtr->imageBuffer)
+    delete this->dataPtr->imageBuffer;
+}
+
+/////////////////////////////////////////////////
+bool BoundingBoxCameraSensor::Init()
+{
+  return CameraSensor::Init();
+}
+
+/////////////////////////////////////////////////
+bool BoundingBoxCameraSensor::Load(sdf::ElementPtr _sdf)
+{
+  sdf::Sensor sdfSensor;
+  sdfSensor.Load(_sdf);
+  return this->Load(sdfSensor);
+}
+
+/////////////////////////////////////////////////
+bool BoundingBoxCameraSensor::Load(const sdf::Sensor &_sdf)
+{
+  std::lock_guard<std::mutex> lock(this->dataPtr->mutex);
+
+  // BoundingBox Type
+  sdf::ElementPtr sdfElement = _sdf.Element();
+  if (sdfElement->HasElement("box_type"))
+  {
+    std::string type = sdfElement->Get<std::string>("box_type");
+
+    // convert type to lowercase
+    std::for_each(type.begin(), type.end(), [](char & c){
+      c = std::tolower(c);
+    });
+
+    if (type == "full_2d" || type == "full_box_2d")
+      this->dataPtr->type = rendering::BoundingBoxType::FullBox2D;
+    else if (type == "2d" || type == "visible_2d"
+      || type == "visible_box_2d")
+      this->dataPtr->type = rendering::BoundingBoxType::VisibleBox2D;
+    else if (type == "3d")
+      this->dataPtr->type = rendering::BoundingBoxType::Box3D;
+    else
+    {
+      ignerr << "Unknown bounding box type " << type << std::endl;
+      return false;
+    }
+  }
+
+  if (!Sensor::Load(_sdf))
+  {
+    return false;
+  }
+
+  // Check if this is the right type
+  if (_sdf.Type() != sdf::SensorType::BOUNDINGBOX_CAMERA)
+  {
+    ignerr << "Attempting to a load a BoundingBox Camera sensor, but received "
+      << "a " << _sdf.TypeStr() << std::endl;
+    return false;
+  }
+
+  if (_sdf.CameraSensor() == nullptr)
+  {
+    ignerr << "Attempting to a load a BoundingBox Camera sensor, but received "
+      << "a null sensor." << std::endl;
+    return false;
+  }
+
+  this->dataPtr->sdfSensor = _sdf;
+
+  this->dataPtr->topicBoundingBoxes = this->Topic();
+  this->dataPtr->topicImage = this->Topic() + "_image";
```

Comment:
I wonder if it would be better to use `/image` as the suffix instead of `_image`. I will check with other people on the team and follow-up on this, so let's leave this open for now.